src/main/java/seedu/address/logic/commands/friends/EditFriendCommand.java

```diff
@@ -0,0 +1,153 @@
+package seedu.address.logic.commands.friends;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.FLAG_FRIEND_NAME;
+
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.friend.Friend;
+import seedu.address.model.friend.FriendId;
+import seedu.address.model.friend.FriendName;
+import seedu.address.model.gamefriendlink.GameFriendLink;
+
+/**
+ * Edits the details of an existing friend in the address book.
+ */
+public class EditFriendCommand extends Command {
+
+    public static final String COMMAND_WORD = "--edit";
+    public static final String USAGE_EXAMPLE = "Example: friend " + COMMAND_WORD + " Draco "
+            + FLAG_FRIEND_NAME + "Marcus Tang";
+    public static final String MESSAGE_EDIT_FRIEND_SUCCESS = "Friend edited - %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided. \n" + USAGE_EXAMPLE;
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the friend identified "
+            + "by friend's FRIEND_ID. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: FRIEND_ID "
+            + FLAG_FRIEND_NAME + "NEW_FRIEND_NAME \n"
+            + USAGE_EXAMPLE;
```

Comment:
Consider shortening the message as it causes it is a lot of text and cannot be seen at a glance without scrolling. I feel it might be better to just give a simple example, if the user needs more help, they can use the `help` command instead.
![image](https://user-images.githubusercontent.com/50147457/138324334-2cd95e9d-c3e9-4241-aa1a-abb16d4e5e66.png)